File: evaluation-system-backend/src/main/java/com/clt/evaluation_system_backend/model/Emp.java
package com.clt.evaluation_system_backend.model;

import java.time.LocalDateTime;

import com.clt.evaluation_system_backend.annotation.SeqTable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@SeqTable("emp")
public class Emp {
    private String empId;
    private String empNm;
    private String empNo;
    private String empEmail;

    private String compRoleCd;
    private String deptCd;
    private String posCd;
    private String lvlCd;
    private Double salaryLvl;
    private String teamId;
    private String empStatusCd;

    private LocalDateTime last_rev_dt;
    private LocalDateTime next_rev_dt;

    private String creUsrId;
    private LocalDateTime creDt;
    private String updUsrId;
    private LocalDateTime updDt;
    private String delFlg;
    private FormSubm formSubm;
}

File: evaluation-system-frontend/src/types/emp.ts
export interface EmployeeBackend {
  empId: string;
  empNm: string;
  empNo: string;
  empEmail: string;

  compRoleCd: string;
  deptCd: string;
  posCd: string;
  lvlCd: string;

  salaryLvl: number;
  teamId: string | null;
  empStatusCd: string;

  formSubm?: FormSubmBackend | null;
}
export interface FormSubmBackend {
  formSubmId: string;
  formId: string;
  empId: string;
  empNm: string;
  empNo: string;

  empCurrDeptCd: string;
  empCurrPosCd: string;
  empCurrLvlCd: string;

  revDt: string;      
  nextRevDt: string;  
  formSubmStatus: string;

  form?: FormBackend | null;
  bossRevList?: BossRevBackend[];
}
export interface FormBackend {
  formId: string;
  formTitle: string;
  formVer: number;
  deptCd: string;
  posCd: string;
  lvlCd: string;
  formStatus: string;
}
export interface BossRevBackend {
  bossRevId: number;
  formSubmId: string;
  empNo: string;

  bossNo: string;
  bossRevRole: string;
  bossRevOrdNo: number;
  isFinal: string;

  bossEmp?: BossEmployeeBackend | null;
}
export interface BossEmployeeBackend {
  empId: string;
  empNm: string;
  empNo: string;
  empEmail: string;

  compRoleCd: string;
  deptCd: string;
  posCd: string;
  lvlCd: string;

  salaryLvl: number;
  teamId: string | null;
  empStatusCd: string;
}

export const BossRevRoleEnum = {
  LEADER: "LEADER",
  GM: "GM",
  DIRECTOR: "DIRECTOR",
} as const;
